app/src/main/java/kth/jjve/xfran/CalendarViewActivity.java

```diff
@@ -78,7 +81,7 @@ protected void onCreate(Bundle savedInstanceState) {
         });
 
         buttonNewEvent.setOnClickListener(v -> {
-            startActivity(new Intent(this, PlanEventActivity.class));
+            startActivity(new Intent(this, PlanEventActivity.class)); // TODO delete this probs!!!!
             Intent intent = new Intent(this, PlanEventActivity.class);
             intent.putExtra(WORKOUT_ID, position);
             intent.putExtra(WORKOUT_OBJ, mWorkout);
```

Comment:
You first start an activity from an intent and then you create a new intent? Seems like this could be why the activity opens twice. So yeah, before merging, I would suggest that you delete line 84 and add a line after 87 that says:
`startActivity(intent);`